redash/handlers/query_results.py

```diff
@@ -154,6 +156,23 @@ def options(self, query_id=None, query_result_id=None, filetype='json'):
 
         return make_response("", 200, headers)
 
+    def _fetch_rows(self, query_id):
+        query = models.Query.get_by_id_and_org(query_id, self.current_org)
+        query_result = models.QueryResult.get_by_id_and_org(query.latest_query_data_id, self.current_org)
+
+        require_access(query_result.data_source.groups, self.current_user, view_only)
```

Comment:
Better to do this before we went through the trouble of loading the query and query_result.

Reply:
The best I can think of here is moving this one line up and running ```require_access(query.data_source.groups, self.current_user, view_only)```, otherwise I can't see how we get the groups to require access to.

Reply:
I overlooked the fact it needs to know the data source, so yes -- let's move it before query_result.